ios/RCTContacts/RCTContacts.m

```diff
@@ -61,41 +61,132 @@ - (NSDictionary *)constantsToExport
+                                   )
 {
     CNContactStore *contactStore = [[CNContactStore alloc] init];
     if (!contactStore)
         return;
-    [self getContactsFromAddressBook:contactStore matchingString:string callback:callback];
+    [self getContactsFromAddressBook:contactStore matchingString:string withThumbnails:withThumbnails callback:callback];
 }
 
 -(void) getContactsFromAddressBook:(CNContactStore *)store
                     matchingString:(NSString *)searchString
+                    withThumbnails:(BOOL) withThumbnails
                           callback:(RCTResponseSenderBlock)callback
 {
-    NSMutableArray *contacts = [[NSMutableArray alloc] init];
+    NSMutableArray<CNContact*> *contacts = [[NSMutableArray alloc] init];
     NSError *contactError = nil;
-    NSArray *keys = @[
-                      CNContactEmailAddressesKey,
-                      CNContactPhoneNumbersKey,
-                      CNContactFamilyNameKey,
-                      CNContactGivenNameKey,
-                      CNContactMiddleNameKey,
-                      CNContactPostalAddressesKey,
-                      CNContactOrganizationNameKey,
-                      CNContactJobTitleKey,
-                      CNContactImageDataAvailableKey,
-                      CNContactNoteKey,
-                      CNContactUrlAddressesKey,
-                      CNContactBirthdayKey
-                      ];
-    NSArray *arrayOfContacts = [store unifiedContactsMatchingPredicate:[CNContact predicateForContactsMatchingName:searchString]
-                                                           keysToFetch:keys
-                                                                 error:&contactError];
-    [arrayOfContacts enumerateObjectsUsingBlock:^(CNContact * _Nonnull obj, NSUInteger idx, BOOL * _Nonnull stop) {
-        NSDictionary *contactDictionary = [self contactToDictionary:obj withThumbnails:NO];
-        [contacts addObject:contactDictionary];
+
+    NSArray *nameKeys = @[
+        CNContactFamilyNameKey,
+        CNContactGivenNameKey,
+        CNContactMiddleNameKey,
+        CNContactNamePrefixKey,
+        CNContactNameSuffixKey,
+        CNContactPreviousFamilyNameKey,
+        CNContactNicknameKey,
+        CNContactPhoneticGivenNameKey,
+        CNContactPhoneticMiddleNameKey,
+        CNContactPhoneticFamilyNameKey
+        ];
+
+    NSMutableArray *keys = [[NSMutableArray alloc]init];
+    [keys addObjectsFromArray:nameKeys];
+    [keys addObjectsFromArray:@[
+        CNContactEmailAddressesKey,
+        CNContactPhoneNumbersKey,
+        CNContactPostalAddressesKey,
+        CNContactOrganizationNameKey,
+        CNContactJobTitleKey,
+        CNContactImageDataAvailableKey,
+        CNContactNoteKey,
+        CNContactUrlAddressesKey,
+        CNContactBirthdayKey
+        ]];
+
+    if (withThumbnails) {
+        [keys addObject:CNContactThumbnailImageDataKey];
```

Comment:
I'd grab this regardless, since the expensive part is copying the data, not being aware of it.

Reply:
Great, that means I don't need to change the signature. I can just had `withThumbnails: YES` when adding the contact

Reply:
Yup